executor/builder.go

```diff
@@ -1974,15 +1972,33 @@ func (b *executorBuilder) buildMemTable(v *plannercore.PhysicalMemTable) Executo
 			}
 		case strings.ToLower(infoschema.TableStatementsSummary),
 			strings.ToLower(infoschema.TableStatementsSummaryHistory),
+			strings.ToLower(infoschema.TableStatementsSummaryEvicted),
+			strings.ToLower(infoschema.ClusterTableStatementsSummary),
 			strings.ToLower(infoschema.ClusterTableStatementsSummaryHistory),
-			strings.ToLower(infoschema.ClusterTableStatementsSummary):
+			strings.ToLower(infoschema.ClusterTableStatementsSummaryEvicted):
+			var extractor *plannercore.StatementsSummaryExtractor
+			if v.Extractor != nil {
+				extractor = v.Extractor.(*plannercore.StatementsSummaryExtractor)
+			}
```

Comment:
Beside the difference between `stmtSummaryRetrieverV2` and `stmtSummaryRetriever`, there are other changes:
1. moved tables
2. the nil check for `v.Extractor`

Feeling curious about these.

Reply:
1. Previously `TableStatementsSummaryEvicted` and `ClusterTableStatementsSummaryEvicted` were handled separately, and the code organization was different from other tables. Here we just group them together in `stmtsummary.go`.
2. I constructed the case where v.Extractor is nil in my tests, and I found that `v.Extractor.(*plannercore.StatementsSummaryExtractor)` will panic due to a type mismatch.

Reply:
Previously, to retrieve `TableStatementsSummaryEvicted` and `ClusterTableStatementsSummaryEvicted` we use `memtableRetriever` but `stmtSummaryRetriever` here. Do they act the same?

Reply:
Yes, I just migrated this part of code(just a bit: https://github.com/pingcap/tidb/pull/40814/files#diff-feafb829b9dc329222267c9010f60ed7ce8800012c1e6dbcf6beb27d99e71b49L2308-L2323) to `executor/stmtsummary.go`. I think this will make stmtsummary code more independent.